lib/projects/upload.ts

```diff
@@ -147,11 +162,16 @@ export async function handleProjectUpload<T>(
         }
       }
 
+      await ensureProjectExists(accountId, projectConfig.name, {
+        forceCreate,
+        uploadCommand: isUploadCommand,
+      });
```

Comment:
Moving these here assures that we don't call it until we have validated the project.